plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/cache/TestMutableDeltaLogDeltaLakeCacheKeyProvider.java

```diff
@@ -0,0 +1,69 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.deltalake.cache;
+
+import io.trino.filesystem.Location;
+import io.trino.filesystem.TrinoInputFile;
+import io.trino.filesystem.cache.CacheKeyProvider;
+import io.trino.filesystem.memory.MemoryInputFile;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+import org.junit.jupiter.api.parallel.Execution;
+
+import java.io.IOException;
+import java.util.Optional;
+
+import static io.airlift.slice.Slices.EMPTY_SLICE;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.TestInstance.Lifecycle.PER_CLASS;
+import static org.junit.jupiter.api.parallel.ExecutionMode.CONCURRENT;
+
+@TestInstance(PER_CLASS)
+@Execution(CONCURRENT)
+public class TestMutableDeltaLogDeltaLakeCacheKeyProvider
```

Comment:
Instead of this test,  could we extend https://github.com/trinodb/trino/blob/master/plugin/trino-delta-lake/src/test/java/io/trino/plugin/deltalake/TestDeltaLakeAlluxioCacheFileOperations.java, perhaps by adding an additional catalog with the configuration property set? Though, I'll defer to the other reviewers on this.

Reply:
Should I create a new class that extends AbstractTestQueryFramework? I see that the delta properties are set in createQueryRunner() and I think this new configuration option will break the existing tests that try to check the caching of delta log files.

Reply:
I added the TestDeltaLakeAlluxioCacheFileOperationsMutableTransactionLog class that extends TestDeltaLakeAlluxioCacheFileOperations.